Remotion/Development/UnitTesting.Compilation/Roslyn/Util/RoslynCompilationUtil.cs

```diff
@@ -0,0 +1,187 @@
+            }
+
+            //
+            // UseAspNetSettings - default true. This was meant to be an ASP.Net support package first and foremost.
+            //
+            bool useAspNetSettings = true;
+            if (options.TryGetValue("UseAspNetSettings", out string sUANS))
+            {
+                // Failure to parse sets to 'false', but we want to keep the default 'true'.
+                if (!Boolean.TryParse(sUANS, out useAspNetSettings))
+                    useAspNetSettings = true;
+            }
+
+            RoslynProviderOptions providerOptions = new RoslynProviderOptions() {
+                CompilerFullPath = compilerFullPath,
+                CompilerServerTimeToLive = ttl,
+                CompilerVersion = compilerVersion,
+                WarnAsError = warnAsError,
+                UseAspNetSettings = useAspNetSettings,
+                AllOptions = options
+            };
+
+            return providerOptions;
+        }
+
+        internal static IDictionary<string, string> GetProviderOptionsCollection(string fileExt)
+        {
+            Dictionary<string, string> opts = new Dictionary<string, string>();
+
+            if (!CodeDomProvider.IsDefinedExtension(fileExt))
+                return new ReadOnlyDictionary<string, string>(opts);
+
+            CompilerInfo ci = CodeDomProvider.GetCompilerInfo(CodeDomProvider.GetLanguageFromExtension(fileExt));
+
+            if (ci == null)
+                return new ReadOnlyDictionary<string, string>(opts);
+
+            // There is a fun little comment about this property in the framework code about making it
+            // public after 3.5. Guess that didn't happen. Oh well. :)
+            PropertyInfo pi = ci.GetType().GetProperty("ProviderOptions",
+                BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.IgnoreCase | BindingFlags.Instance);
+            if (pi == null)
+                return new ReadOnlyDictionary<string, string>(opts);
+
+            return new ReadOnlyDictionary<string, string>((IDictionary<string, string>)pi.GetValue(ci, null));
+        }
+
+        internal static void PrependCompilerOption(CompilerParameters compilParams, string compilerOptions)
+        {
+            if (compilParams.CompilerOptions == null)
+            {
+                compilParams.CompilerOptions = compilerOptions;
+            }
+            else
+            {
+                compilParams.CompilerOptions = compilerOptions + " " + compilParams.CompilerOptions;
+            }
+        }
+
+        internal static string CompilerFullPath(string relativePath)
+        {
+            string compilerFullPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, relativePath);
+            return compilerFullPath;
+        }
+
+        internal static bool IsDebuggerAttached
+        {
+            get {
+                return IsDebuggerPresent() || Debugger.IsAttached;
+            }
+        }
+
+        [DllImport("kernel32.dll")]
```

Comment:
Meh, not something we fix now, but in the future this might bite us if we do linux support. Of course, then we also need a Linux-based roslyn or just can't do those tests in Linux...